src/main/java/frc/robot/subsystems/LogSubsystem.java

```diff
@@ -0,0 +1,153 @@
+package frc.robot.subsystems; 
+ 
+import java.io.File;
+import java.io.IOException;
+import java.io.Writer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+ 
+import edu.wpi.first.wpilibj.DriverStation;
+ 
+public class LogSubsystem {
+    public String m_subsystemName;
+    private static LogSubsystem INSTANCE = new LogSubsystem();
+    public String m_level;
+    public static final String RED = "\033[0;31m";     // RED
+    public static final String BLUE = "\033[0;34m";    // BLUE
+    
+    public static LogSubsystem getInstance() {
+        return INSTANCE;
+    }
+ 
+ 
+    public enum Level {
+        kRobot, kInfo, kWarning, kLibrary
+    }
+    private final List<LogSource> dataSources = new ArrayList<>();
+    private Path file;
+ 
+    private String loggingLocation = "/home/lvuser/logs/";
```

Comment:
Şurası için de simülasyon ile çalışıyorsa dosyayı kullanıcının masaüstüne kaydedecek şekilde, simülasyon değilse buradaki gibi bırakırsan hoş olur.